types.go

```diff
@@ -445,45 +445,38 @@ func sprintName(s string) string {
 	var dst strings.Builder
 
 	for i := 0; i < len(s); {
-		if i+1 < len(s) && s[i] == '\\' && s[i+1] == '.' {
```

Comment:
Can you explain the change to this function for me? I can’t quite figure it out.

Reply:
Basically, it's fixing the handling of backslash escapes (including `\046`, an escaped dot within a DNS label) and removing inconsistency between code paths that use the `dst` string builder and those that don't. Specifically:
1. It removes special treatment of `\.`, instead handling that like every other backslash escape.
2. It fixes the incorrect conversion of `\046` (returned from `nextByte` as `'.', 4`) into an unescaped `.`. Note that the existing incorrect behavior is only visible when the string builder has already been used, because `dst.Len() != 0` guards the `dst.WriteByte('.')` that appends the unescaped dot.
3. It breaks out of the loop upon encountering a dangling backslash escape sequence (e.g., ```foo\```), excluding it from output (which previously would only happen when the string builder had been used).
4. It updates the backslash-prefixing logic to use the single source of truth.